sdk/tests/wallet/common/mod.rs

```diff
@@ -79,20 +83,87 @@ pub(crate) async fn make_ledger_nano_wallet(
     Ok(wallet_builder.finish().await?)
 }
 
-/// Request funds from the faucet and sync the wallet.
+/// Create an implicit account creation address, request funds from the faucet to it, transition it to an account and
+/// wait until enough mana is generated to send a transaction.
 #[allow(dead_code)]
 pub(crate) async fn request_funds(wallet: &Wallet) -> Result<(), Box<dyn std::error::Error>> {
+    request_funds_from_faucet(FAUCET_URL, &wallet.implicit_account_creation_address().await?).await?;
     request_funds_from_faucet(FAUCET_URL, &wallet.address().await).await?;
 
     // Continue only after funds are received
-    for _ in 0..30 {
+    for i in 0..30 {
+        tokio::time::sleep(std::time::Duration::from_secs(2)).await;
+        wallet
+            .sync(Some(SyncOptions {
+                sync_implicit_accounts: true,
+                ..Default::default()
+            }))
+            .await?;
+        if wallet.ledger().await.implicit_accounts().next().is_some() {
+            break;
+        }
+        if i == 29 {
+            panic!("Faucet no longer wants to hand over coins");
+        }
+    }
+
+    let wallet_ledger = wallet.ledger().await;
+    let implicit_account = wallet_ledger
+        .implicit_accounts()
+        .next()
+        .expect("No implicit account found")
+        .clone();
+    drop(wallet_ledger);
+
+    let mut tries = 0;
+    while let Err(ClientError::Node(iota_sdk::client::node_api::error::Error::NotFound(_))) = wallet
+        .client()
+        .get_account_congestion(&AccountId::from(&implicit_account.output_id), None)
+        .await
+    {
+        tries += 1;
         tokio::time::sleep(std::time::Duration::from_secs(2)).await;
-        let balance = wallet.sync(None).await?;
-        if balance.base_coin().available() > 0 {
-            return Ok(());
+        if tries > 100 {
+            panic!("Can't get_account_congestion for implicit account");
         }
     }
-    panic!("Faucet no longer wants to hand over coins");
+
+    let transaction = wallet
+        .implicit_account_transition(
+            &implicit_account.output_id,
+            BlockIssuerKeySource::ImplicitAccountAddress,
+        )
+        .await?;
+
+    wallet
+        .wait_for_transaction_acceptance(&transaction.transaction_id, None, None)
+        .await?;
+
+    wallet.sync(None).await?;
+
+    // Is this better than using the congestion endpoint?
+    // prepare a big tx and wait the time it takes until enough mana is generated
+    #[allow(unused_variables)]
+    if let Err(WalletError::Client(ClientError::TransactionBuilder(TransactionBuilderError::InsufficientMana {
+        found,
+        required,
+        slots_remaining,
+    }))) = wallet
+        .prepare_send(vec![SendParams::new(1_000_000, wallet.address().await)?; 10], None)
+        .await
+    {
+        tokio::time::sleep(std::time::Duration::from_secs(
```

Comment:
How exactly does it help? You aren't starting at exactly the start of a slot, so if you wait for the full duration of a slot it can be delayed quite a few seconds

Reply:
Yes sorry, I only looked for `tokio::time::sleep(std::time::Duration::from_secs(` which is also used above with 2 seconds
Changed